packages/sample-app/src/components/common/LoadPluginModal.tsx

```diff
@@ -1,38 +1,49 @@
 import { usePluginStore } from '@openshift/dynamic-plugin-sdk';
-import { Button, Modal, Text, TextContent, TextInput } from '@patternfly/react-core';
+import { Button, Checkbox, Form, FormGroup, Modal, TextInput } from '@patternfly/react-core';
 import * as React from 'react';
 import { isValidURL } from '../../utils';
 
-const isValidPluginBaseURL = (value: string) => isValidURL(value) && value.endsWith('/');
-
 type LoadPluginModalProps = {
-  defaultBaseURL?: string;
+  defaultManifestURL?: string;
 };
 
 export type LoadPluginModalRefProps = {
   open: VoidFunction;
 };
 
 const LoadPluginModal = React.forwardRef<LoadPluginModalRefProps, LoadPluginModalProps>(
-  ({ defaultBaseURL = 'http://localhost:9001/' }, ref) => {
+  ({ defaultManifestURL = 'http://localhost:9001/plugin-manifest.json' }, ref) => {
     const [isModalOpen, setModalOpen] = React.useState(false);
-    const [baseURL, setBaseURL] = React.useState(defaultBaseURL);
-    const [baseURLValid, setBaseURLValid] = React.useState(isValidPluginBaseURL(defaultBaseURL));
+
+    const [manifestURL, setManifestURL] = React.useState(defaultManifestURL);
+    const [manifestURLValid, setManifestURLValid] = React.useState(isValidURL(defaultManifestURL));
+    const [forceReload, setForceReload] = React.useState(false);
 
     const pluginStore = usePluginStore();
 
-    const onBaseURLChange = React.useCallback(
+    const onManifestURLChange = React.useCallback(
       (value: string) => {
-        setBaseURL(value);
-        setBaseURLValid(isValidPluginBaseURL(value));
+        setManifestURL(value);
+        setManifestURLValid(isValidURL(value));
       },
-      [setBaseURL, setBaseURLValid],
+      [setManifestURL, setManifestURLValid],
     );
 
-    const loadPlugin = React.useCallback(() => {
-      pluginStore.loadPlugin(baseURL);
+    const onForceReloadChange = React.useCallback(
+      (value: boolean) => {
+        setForceReload(value);
+      },
+      [setForceReload],
+    );
+
+    const closeModal = React.useCallback(() => {
       setModalOpen(false);
-    }, [baseURL, pluginStore, setModalOpen]);
+    }, [setModalOpen]);
+
+    const loadPlugin = React.useCallback(() => {
+      pluginStore.loadPlugin(manifestURL, forceReload);
```

Comment:
@vojtechszocs would it be possible for us to add another scenario where we provide the manifest itself (over the URL) to test that code-path as well?

Reply:
Sounds good to me, the only question is what would be the best way to represent the manifest object input?

Aside from doing this via UI, we could always expose `pluginStore` via the global `window` object, so developers can interact with it directly, for example:
```ts
pluginStore.loadPlugin('http://localhost:9001/plugin-manifest.json');
pluginStore.getPluginInfo();
```

FYI, OpenShift Console does this for non-production builds ([link](https://github.com/openshift/console/blob/7500dbd9fe588c7e83e92f79d9ba32bbd61dc8e3/frontend/public/plugins.ts#L38)).

This could make local testing via the sample application a bit easier.

Reply:
It might be nice to have a UI option to enter the object (for discoverability purposes). Perhaps code editor (https://www.patternfly.org/v4/components/code-editor) or multi line inputs (https://www.patternfly.org/v4/components/text-input) with the prefilled JSON? I do think it's nice to expose the pluginStore for debugging non-prod builds too. 

We can follow up in another PR for this IMO.

Reply:
Chatted with Vojtech and we will follow up in another PR with exposing the plugin loader/store in the sample app.